README.md

```diff
@@ -94,53 +66,79 @@ Each graphics API may have unique requirements that must be satisfied before a c
+
+```js
+const projectionLayer = xrGfx.requestProjectionLayer();
+const quadLayer = xrGfx.requestQuadLayer({ pixelWidth: 1024, pixelHeight: 1024 });
+
+xrSession.updateRenderState({ layers: [projectionLayer, quadLayer] });
 ```
 
-Passing `true` for stereo here indicates that you are able to provide stereo imagery for this layer source, but if the XR device is unable to display stereo imagery it may automatically force the layer to be created as mono instead to reduce memory and rendering overhead. Layers that are created as mono will never be automatically changed to stereo, regardless of hardware capabilities.
+`updateRenderState()` _may_ throw an exception if more layers are specified than the `XRSession` supports simultaneously.
 
 ## Rendering
 
-During `XRFrame` processing each layer source can be updated with new imagery. Calling `getViewSubImage()` with a view from the `XRFrame` will return an `XRSubImage` indicating what section of the associated graphics resources will be presented to the view's associated physical display. If a layer source has not yet been set for the layer `getSubImage()` with return an `XRSubImage` with all values set to zero or false.
+During `XRFrame` processing each layer can be updated with new imagery. Calling `getViewSubImage()` with a view from the `XRFrame` will return an `XRSubImage` indicating what section of the associated graphics resources will be presented to the view's associated physical display. If a layer source has not yet been set for the layer `getSubImage()` with return an `XRSubImage` with all values set to zero or false.
 
-Every layer source will also have a `getCurrent_____()` method that retrieves the graphics resources that should be rendered to for the current `XRFrame`. Calling this method indicates that the imagery should be updated, otherwise any previous rendering will be displayed again, reprojected if necessary.
+Every `XRGraphicsBinding` instance will also have a `getCurrent_____()` method that retrieves the graphics resources associated with a particular layer that should be rendered to for the current `XRFrame`. Calling this method indicates that the imagery should be updated, otherwise any previous rendering will be displayed again, reprojected if necessary.
 
 ```js
 // Render Loop for a projection layer with a WebGL framebuffer source.
 const xrGfx = new XRWebGLGraphicsBinding(xrSession, gl);
-
-const layer = xrSession.requestLayer('projection');
-const source = await xrGfx.setFramebufferLayerSource(layer);
+const layer = xrGfx.requestProjectionLayer();
 
 xrSession.updateRenderState({ layers: [layer] });
 xrSession.requestAnimationFrame(onXRFrame);
 
 function onXRFrame(time, xrFrame) {
   xrSession.requestAnimationFrame(onXRFrame);
 
-  gl.bindFramebuffer(gl.FRAMEBUFFER, source.getCurrentFramebuffer());
+  gl.bindFramebuffer(gl.FRAMEBUFFER, xrGfx.getCurrentFramebuffer(layer));
```

Comment:
How is it known that there IS a current framebuffer? I don't see that it was requested anyhow....

Reply:
The fact that it has a framebuffer backing it is due to the fact that the `XRWebGLGraphicsBinding` is the thing that created the layer. I think that interface name is unnecessarily confusing things though. If it was called, for example, `XRWebGLFramebufferLayerFactory` that would probably help with comprehension of how this pattern is intended to work.